test/general/lambda_naming.pass.cpp

```diff
@@ -30,7 +30,6 @@ using namespace TestUtils;
 
 // This is the simple test for compilation only, to check if lambda naming works correctly
 int main() {
-#if __SYCL_UNNAMED_LAMBDA__ && _ONEDPL_BACKEND_SYCL
```

Comment:
Let's keep it since it guards calls like `sycl::buffer`, which will not work on e.g. `gcc`.